packages/create-sitecore-jss/src/templates/nextjs-sxa/src/pages/api/sitemap.ts

```diff
@@ -0,0 +1,34 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import config from 'temp/config';
+import { GraphQLSitemapService } from '@sitecore-jss/sitecore-jss/site';
+
+const sitemapApi = async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+  const { query: { id } } = req;
+  const currentUrl = process.env.PUBLIC_URL;
+  // create sitemap graphql service
+  const sitemapService = new GraphQLSitemapService({
+    endpoint: config.graphQLEndpoint,
+    apiKey: config.sitecoreApiKey,
+    siteName: config.jssAppName,
+  });
+
+  const existSitemap = await sitemapService.getExistsSitemap(id as string);
+  if (existSitemap) {
+    const sitemapResult = await fetch(`${currentUrl}${existSitemap}` as string, {
```

Comment:
Need to check if `existSitemap` is absolute URL. When published to Experience Edge, I believe the sitemap value with be an absolute URL (using CDN) and in that case we wouldn't append `currentUrl`. It will only be relative on preview schema. And in this case, we should use `config.sitecoreApiHost` instead of public URL. This is what we use for media item rewrites already in default nextjs rewrites.